calyx-py/calyx/builder.py

```diff
@@ -1471,23 +1514,37 @@ def static_seq(*args) -> ast.StaticSeqComp:
     return ast.StaticSeqComp([as_control(x) for x in args])
 
 
-def add_comp_params(comp: ComponentBuilder, input_ports: List, output_ports: List):
+def add_comp_ports(comp: ComponentBuilder, input_ports: List, output_ports: List):
     """
     Adds `input_ports`/`output_ports` as inputs/outputs to comp.
     `input_ports`/`output_ports` should contain an (input_name, input_width) pair.
+    Or they should contain an (input_name, input_width, attributes) triple.
```

Comment:
Again, perhaps this would be a little simpler if we just always required it to be a 3-tuple, and just allowed the last one to be empty? (I realize that this would break clients that are currently proving 2-tuples, but I think that is OK… we don't make a lot of guarantees for API stability in this library, and perhaps breakage is preferable to allowing multiple options here, just for simplicity's sake.)

Reply:
I do think there's something to be said w.r.t ease of use when creating components and their ports and not having to think about empty attribute lists. I [refactored this a bit](https://github.com/calyxir/calyx/pull/2069/files#diff-1b6ee22a9e8b557927fd938de3186a9db39b2efd86dd63605f498edeb3d75160R1522-R1538) to try and make things nicer/remove some code duplication while still allowing for both forms.